internal/armada/scheduling/lease.go

```diff
@@ -276,22 +303,33 @@ func (c *leaseContext) leaseJobs(
 		}
 
 		candidates := make([]*api.Job, 0)
-		candidatesLimit := NewLeasePayloadLimit(limit.remainingJobCount, limit.remainingPayloadSizeLimitBytes, limit.maxExpectedJobSizeBytes)
+		candidatesLimit := newLeasePayloadLimit(limit.remainingJobCount, limit.remainingPayloadSizeLimitBytes, limit.maxExpectedJobSizeBytes)
 		candidateNodes := map[*api.Job]nodeTypeUsedResources{}
 		consumedNodeResources := nodeTypeUsedResources{}
 
 		for _, job := range topJobs {
 			requirement := common.TotalJobResourceRequest(job).AsFloat()
 			remainder = slice.DeepCopy()
 			remainder.Sub(requirement)
-			if isLargeEnough(job, c.minimumJobSize) && remainder.IsValid() && candidatesLimit.IsWithinLimit(job) {
-				newlyConsumed, ok, _ := matchAnyNodeTypeAllocation(job, c.nodeResources, consumedNodeResources)
+
+			if isJobSchedulable(c, job, remainder, candidatesLimit) {
+				if hasPriorityClass(job.PodSpec) {
+					validateOrDefaultPriorityClass(job.PodSpec, c.schedulingConfig.Preemption)
+				}
+				newlyConsumed, ok, err := matchAnyNodeTypeAllocation(
+					job,
+					c.nodeResources,
+					consumedNodeResources,
+					c.schedulingConfig.Preemption.PriorityClasses,
+				)
 				if ok {
 					slice = remainder
 					candidates = append(candidates, job)
 					candidatesLimit.RemoveFromRemainingLimit(job)
 					candidateNodes[job] = newlyConsumed
 					consumedNodeResources.Add(newlyConsumed)
+				} else {
+					log.WithError(err).Error("failed to match node")
```

Comment:
I don't think we want this log line, or at the very least not as an error

I recently removed it from master as it spams the logs with errors but no real details. However that "error" isn't an error and purely is saying that the job doesn't fit onto the existing nodes.
 - This happens a lot, especially on clusters with multiple node types.